streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -456,9 +456,9 @@ StandbyTask createTask(final Consumer<byte[], byte[]> consumer,
             final String logPrefix = threadIdPrefix + String.format("%s [%s] ", "standby-task", taskId);
             final LogContext logContext = new LogContext(logPrefix);
 
-            final ProcessorTopology topology = builder.build(taskId.topicGroupId);
+            final ProcessorTopology topology = builder.buildSubtopology(taskId.topicGroupId);
 
-            if (!topology.stateStores().isEmpty() && !topology.storeToChangelogTopic().isEmpty()) {
```

Comment:
There was actually a bug in this condition: the `storeToChangelogTopic` map actually corresponds to the _entire_ topology, not just this task's subtopology. See changes in `ProcessorTopology` above